File: app/src/main/java/com/musicplayer/model/Song.java
package com.musicplayer.model;

public class Song {
    private String id;
    private String title;
    private String artist;
    private String url;
    private String streamUrl;
    private String thumbnailUrl;
    private long duration;
    private boolean isLocal = false;
    
    public Song(String id, String title, String artist, String url, long duration, String thumbnailUrl) {
        this.id = id;
        this.title = title;
        this.artist = artist;
        this.url = url;
        this.duration = duration;
        this.thumbnailUrl = thumbnailUrl;
    }

    public String getId() { return id; }
    public String getTitle() { return title; }
    public String getArtist() { return artist; }
    public void setArtist(String artist) { this.artist = artist; }
    public String getUrl() { return url; }
    public String getStreamUrl() { return streamUrl; }
    public void setStreamUrl(String streamUrl) { this.streamUrl = streamUrl; }
    public String getThumbnailUrl() { return thumbnailUrl; }
    public long getDuration() { return duration; }
    public boolean isLocal() { return isLocal; }
    public void setLocal(boolean local) { isLocal = local; }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Song song = (Song) o;
        return id != null ? id.equals(song.id) : song.id == null;
    }

    @Override
    public int hashCode() {
        return id != null ? id.hashCode() : 0;
    }
}

File: app/src/main/java/com/musicplayer/model/Song.kt
package com.musicplayer.model

data class Song(
    val id: String,
    val title: String,
    var artist: String,
    val url: String,
    val duration: Long,
    val thumbnailUrl: String
) {
    var streamUrl: String? = null
    var isLocal: Boolean = false
    var isHeader: Boolean = false

    companion object {
        @JvmStatic
        fun createHeader(title: String): Song {
            return Song("header_${System.currentTimeMillis()}", title, "", "", 0, "").apply {
                isHeader = true
            }
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Song) return false
        return id == other.id
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }
}
